java/src/main/java/ai/onnxruntime/OnnxSparseTensor.java

```diff
@@ -0,0 +1,876 @@
+  private final LongBuffer innerIndices;
+  private final Buffer data;
+
+  /**
+   * Construct a sparse tensor from JNI.
+   *
+   * @param nativeHandle The tensor native handle.
+   * @param allocatorHandle The allocator handle.
+   * @param sparseType The sparsity type.
+   * @param info The tensor info.
+   */
+  OnnxSparseTensor(
+      long nativeHandle, long allocatorHandle, SparseTensorType sparseType, TensorInfo info) {
+    this(nativeHandle, allocatorHandle, sparseType, info, null, null, null);
+  }
+
+  /**
+   * Construct a COO or block sparse tensor.
+   *
+   * @param nativeHandle The tensor native handle.
+   * @param allocatorHandle The allocator handle.
+   * @param sparseType The sparsity type.
+   * @param info The tensor info.
+   * @param indices The indices buffer.
+   * @param data The data buffer.
+   */
+  OnnxSparseTensor(
+      long nativeHandle,
+      long allocatorHandle,
+      SparseTensorType sparseType,
+      TensorInfo info,
+      Buffer indices,
+      Buffer data) {
+    this(nativeHandle, allocatorHandle, sparseType, info, indices, null, data);
+  }
+
+  /**
+   * Construct a sparse tensor.
+   *
+   * <p>If the tensor is COO or block sparse then innerIndices may be null.
+   *
+   * @param nativeHandle The tensor native handle.
+   * @param allocatorHandle The allocator handle.
+   * @param sparseType The sparsity type.
+   * @param info The tensor info.
+   * @param indices The indices buffer.
+   * @param innerIndices The inner indices buffer.
+   * @param data The data buffer.
+   */
+  OnnxSparseTensor(
+      long nativeHandle,
+      long allocatorHandle,
+      SparseTensorType sparseType,
+      TensorInfo info,
+      Buffer indices,
+      LongBuffer innerIndices,
+      Buffer data) {
+    super(nativeHandle, allocatorHandle, info);
+    this.sparseTensorType = sparseType;
+    this.indices = indices;
+    this.innerIndices = innerIndices;
+    this.data = data;
+  }
+
+  /**
+   * Creates a Sparse Tensor in ORT from the Java side representation.
+   *
+   * @param env The OrtEnvironment.
+   * @param tensor The Java side representation.
+   * @return The sparse tensor in ORT.
+   * @throws OrtException If the tensor could not be created or was invalid.
+   */
+  public static OnnxSparseTensor createSparseTensor(OrtEnvironment env, SparseTensor tensor)
+      throws OrtException {
+    return createSparseTensor(env, env.defaultAllocator, tensor);
+  }
+
+  static OnnxSparseTensor createSparseTensor(
+      OrtEnvironment env, OrtAllocator allocator, SparseTensor tensor) throws OrtException {
+    if ((!env.isClosed()) && (!allocator.isClosed())) {
+      TensorInfo info = TensorInfo.constructFromSparseTensor(tensor);
+      OnnxJavaType indicesType = tensor.getIndicesType();
+      OrtUtil.BufferTuple indicesTuple = OrtUtil.prepareBuffer(tensor.getIndices(), indicesType);
+      OrtUtil.BufferTuple dataTuple = OrtUtil.prepareBuffer(tensor.getData(), info.type);
```

Comment:
>getData [](http://example.com/codeflow?start=67&length=7)

In sparse terminology this is either Values (vs Indices) or NNZ.

Reply:
I can rename data to values everywhere.

Reply:
I think it would a good thing.

Reply:
Done.